packages/babel-core/src/transformation/index.ts

```diff
@@ -93,22 +94,14 @@ function* transformFile(file: File, pluginPasses: PluginPasses): Handler<void> {
     }
 
     for (const [plugin, pass] of passPairs) {
-      const fn = plugin.pre;
-      if (fn) {
-        // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
-        const result = fn.call(pass, file);
-
-        // If we want to support async .pre
-        yield* [];
-
-        if (isThenable(result)) {
-          throw new Error(
-            `You appear to be using an plugin with an async .pre, ` +
-              `which your current version of Babel does not support. ` +
-              `If you're using a published plugin, you may need to upgrade ` +
-              `your @babel/core version.`,
-          );
-        }
+      if (plugin.pre) {
+        const fn = maybeAsync(
+          plugin.pre,
+          `You appear to be using an async plugin/preset, but Babel has been called synchronously`,
```

Comment:
Also not sure about error message here. Should that be adjusted to notify that exactly pre/post function is async, or that general error message would be enough?

Reply:
This should be enough -- from a user's point of view, it doesn't matter which part exactly of a plugin is async :)